core/client/fs/src/main/java/alluxio/client/file/dora/netty/NettyDataReaderStateMachine.java

```diff
@@ -429,20 +428,9 @@ void pollResponseFromQueue() {
 
   void onReceivedData(ByteBuf buf, Transition<State, TriggerEvent> transition) {
     Preconditions.checkState(TriggerEvent.DATA_AVAILABLE == transition.getTrigger());
-    ByteBuf byteBuf = buf.slice();
-    int readableBytes = byteBuf.readableBytes();
-    int sliceEnd = Math.min(readableBytes, mLength - mBytesRead);
-    // todo(bowen): handle case where ByteBuf does not support getting a bytebuffer
-    ByteBuffer toWrite = byteBuf.nioBuffer(0, sliceEnd);
-    try {
-      mBytesRead += mOutputChannel.write(toWrite);
-    } catch (IOException ioe) {
-      fireNext(mTriggerEventsWithParam.mOutputErrorEvent, ioe);
-      return;
-    } finally {
-      // previously retained in packet read handler
-      byteBuf.release();
-    }
+    int bytesToWrite = buf.readableBytes();
+    mOutputBuffer.writeBytes(buf);
```

Comment:
need to release the `buf` after the read is done.

Reply:
Need to keep the error handling code to account for the case where `mOutputBuffer.writeBytes(buf)`  throws an exception.

Reply:
Addressed, thanks!